### ISSUE TITLE
Ability to configure LACP bonding via Network Config using console

### ISSUE THREAD

**Issue by kvaps** [siderolabs/talos#9708]
## Feature Request

Ability to configure LACP bonding via Network Config using console

### Description

Currently there is no way to configure bonding directly on metal:
![image](https://github.com/user-attachments/assets/06f67307-e038-4163-bd64-68398696549a)

This is blocking us from installing Talos Linux on such enviroments where LACP is enabled by default

**Comment by smira** [comment]
There are quite a few methods available now: https://www.talos.dev/v1.8/talos-guides/install/bare-metal-platforms/network-config/

Moreover, 802.3ad bonds usually allow single-link operations, so you can boot, configure single link enough to submit machine configuration with full bond configuration. This e.g. how Equnix Metal works.

**Comment by kvaps** [comment]
> Moreover, 802.3ad bonds usually allow single-link operations, so you can boot, configure single link enough to submit machine configuration with full bond configuration. This e.g. how Equnix Metal works.

Unfortunately this is not working for some providers, eg servers.com does not supporting this.

**Comment by smira** [comment]
As far I know that _some_ providers are working to provide `nocloud` metadata so that Talos can auto-configure networking, but I can't comment on that myself.

**Comment by vills** [comment]
@kvaps on behalf of mentioned Servers.com provider, i can confirm that we are working on bringing an automated way to install Talos on our platform :-)

**Comment by kvaps** [comment]
Thank you, but I mentioned servers.com just for example. There are a lot of providers and bare-metal configurations with LACP enabled, and it's not so easy to install Talos Linux on them

**Comment by Haribo112** [comment]
To add on to this issue; I thought I could work around it by creating a custom iso image with the bond configured in the kernel arguments. This works to an extent: according to the messages on screen during startup, the bond is created and active. However, I am unable to select this bond as my interface in the Network Config console so now I cannot set an IP address on it. 

I know I can setup the IP address using kernel parameters as well, but then I would need a unique image for every server I plan to use. That would become cumbersome quite quickly.

**Comment by SkalaNetworks** [comment]
There's a window marked "New Config", maybe allowing arbitrary configuration in there (like a text editor) would be enough? It's not just abou LACP, it's also VLANs, routes and more than can be useful to setup using the dashboard. Just the bare minimum to get connectivity up and upload a real config using talosctl.

**Comment by buroa** [comment]
> creating a custom iso image with the bond configured in the kernel arguments

Yep [this is how I do it currently](https://github.com/buroa/k8s-gitops/blob/main/talos/schematic.yaml.j2#L17-L20) however, I think it's quite janky and could be better.

The jank: because dhcp is not running until Talos is started, the linux kernel waits 300 seconds or so for a network link to become active, and then boots into Talos where it sees that parameter and starts dhcp.

**Comment by smira** [comment]
This all should be solved with https://github.com/siderolabs/talos/issues/10959
